src/trix/models/composition.coffee

```diff
@@ -115,11 +120,11 @@ class Trix.Composition extends Trix.BasicObject
       else
         if block.isEmpty()
           @removeLastBlockAttribute()
-        else if block.text.getStringAtRange([endLocation.offset - 1, endLocation.offset]) is "\n" 
+        else if previousCharacter is "\n"
           @breakFormattedBlock()
-        else if @getPositionInBlock() is "end"
+        else if breaksOnReturn and nextCharacter is "\n"
```

Comment:
@sstephenson Is this readable enough that it doesn't need a helper method to reveal the intent now?